gufe/chemicalsystem.py

```diff
@@ -126,6 +126,54 @@ def total_charge(self):
                 total_charge += fc
         return total_charge
 
+    def contains(self, item: Component | type[Component]) -> bool:
+        """Check if a Component or Component type is in this ChemicalSystem.
+
+        Parameters
+        ----------
+        item : Component or type of Component
+            The Component instance or class to check for.
+
+        Returns
+        -------
+        bool
+            Returns True if the item is found in the ChemicalSystem, False otherwise.
```

Comment:
not blocking, but bonus points for adding an `Examples` section in the spirit of gufe usability drive  (below also)